valhalla/baldr/graphtileheader.h

```diff
@@ -232,6 +232,14 @@ class GraphTileHeader {
     nodecount_ = count;
   }
 
+  /**
+   * Does the tile contain OSM ids?
+   * @return \c true if it does and \c false otherwise.
+   */
+  bool has_osmids_for_nodes() const {
+    return osmids_for_nodes_offset_ != 0;
+  }
+
```

Comment:
Are you happy with using the offset like this? In `GraphTileBuilder`, if there are no OSM ids I set the offset to zero. In `GraphTile` I have an `assert(has_osmids_for_nodes())` before indexing the array of OSM ids.

- Should that be a runtime error rather than an assert?
- Might I be causing problem or simply not be being explicit enough by using an offset in a logical condition like this?

Reply:
i think this works perfectly fine. i dont think we need an assertion though. if anything checking for validity should be when we set the pointer in graphtiles initialization, we could check that the number of ids matches what the number of nodes in the tile is `(end_of_tile - ids_offset)/sizeof(uint64_t) == header->nodecount()` or something like that

Reply:
Do we want to add that check for validity?